server/channels/utils/license.go

```diff
@@ -74,6 +74,13 @@ func (l *LicenseValidatorImpl) ValidateLicense(signed []byte) (bool, string) {
 	plaintext := decoded[:len(decoded)-256]
 	signature := decoded[len(decoded)-256:]
 
+	var publicKey []byte
+	switch model.GetExternalServiceEnvironment() {
+	case model.ExternalServiceEnvironmentDefault, model.ExternalServiceEnvironmentCloud:
```

Comment:
What would be different for the cloud environment from the prod environment? It seems that in this PR it's always considered the same case as Default which seems to be the case for Rudder, Licenses and Stripe keys

Reply:
Good question. Somehow, I assumed going in that we needed to distinguish cloud from self-managed in some way, given (I think) we have separate Docker images for each, but you're right that after these changes nothing obvious shook out.

cc @stylianosrigas, any idea what distinguishes these images from your side?

Reply:
Hey @stylianosrigas  @lieut-data 

I apologize for any inconvenience caused by my lack of context. If you could kindly provide some additional information, I would greatly appreciate it. I would like to understand exactly what information is needed here.

Reply:
Hi @phoinixgrr! I'm helping wrap up work that @toninis, Nathaniel, @marianunez and I started to help us realize the delivery team's vision of a single build artifact that conforms to the 12 factor principle, instead of our current practice of a myriad of release pipelines that build various Docker images, e.g. `mattermost-enterprise-edition` for self-managed customers, `mattermost-cloud` for Cloud, and `mm-ee-test` for testing environments.

This PR makes it possible for a single binary to be any of the above with the appropriate environment variable, but after incorporating all necessary changes, we found no need to distinguish between the `mattermost-enterprise-edition` and the `mattermost-cloud`. Do you know why we build separate images in this case? Is there some other difference we need to account for in these application changes? Or is it simply that we publish builds from the `cloud` branch to a different image?

Reply:
(Chiming in)

> Or is it simply that we publish builds from the cloud branch to a different image?

This is what I thought too.

Since the introduction of cloud, in terms of testing, I'm not treating `mattermost-cloud` differently from `mattermost-enterprise-edition` (or their equivalent test images), except for loading cloud license, cloud-specific server config (together with test CWS test, etc) and cloud-specific env variables (api key, customer and installation IDs).